agent/stats/task_windows.go

```diff
@@ -0,0 +1,192 @@
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
+
+	dockerstats "github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+)
+
+// Making it visible for unit testing
+var execCommand = exec.Command
+
+type StatsTask struct {
+	*statsTaskCommon
+}
+
+func newStatsTaskContainer(taskARN string, containerPID string, numberOfContainers int,
+	resolver resolver.ContainerMetadataResolver, publishInterval time.Duration, taskENIs task.TaskENIs) (*StatsTask, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	devices := make([]string, len(taskENIs))
+	for index, device := range taskENIs {
+		devices[index] = device.LinkName
+	}
+
+	return &StatsTask{
+		statsTaskCommon: &statsTaskCommon{
+			TaskMetadata: &TaskMetadata{
+				TaskArn:          taskARN,
+				ContainerPID:     containerPID,
+				DeviceName:       devices,
+				NumberContainers: numberOfContainers,
+			},
+			Ctx:                   ctx,
+			Cancel:                cancel,
+			Resolver:              resolver,
+			metricPublishInterval: publishInterval,
+		},
+	}, nil
+}
+
+func (taskStat *StatsTask) retrieveNetworkStatistics() (map[string]dockerstats.NetworkStats, error) {
+	if len(taskStat.TaskMetadata.DeviceName) == 0 {
+		return nil, errors.Errorf("unable to find any device name associated with the task %s", taskStat.TaskMetadata.TaskArn)
+	}
+
+	networkStats := make(map[string]dockerstats.NetworkStats, len(taskStat.TaskMetadata.DeviceName))
+	for _, device := range taskStat.TaskMetadata.DeviceName {
+		networkAdaptorStatistics, err := taskStat.getNetworkAdaptorStatistics(device)
+		if err != nil {
+			return nil, err
+		}
+		networkStats[device] = *networkAdaptorStatistics
+	}
+
+	return networkStats, nil
+}
+
+// getNetworkAdaptorStatistics returns the network statistics per container for the given network interface.
+func (taskStat *StatsTask) getNetworkAdaptorStatistics(device string) (*dockerstats.NetworkStats, error) {
+	// Ref: https://docs.microsoft.com/en-us/powershell/module/netadapter/get-netadapterstatistics?view=windowsserver2019-ps
+	// The Get-NetAdapterStatistics cmdlet gets networking statistics from a network adapter.
+	// The statistics include broadcast, multicast, discards, and errors.
+	cmd := "Get-NetAdapterStatistics -Name \"" + device + "\" | Format-List -Property *"
+	out, err := execCommand("powershell", "-Command", cmd).CombinedOutput()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to run Get-NetAdapterStatistics for %s", device)
+	}
+	str := string(out)
+
+	lines := strings.Split(str, "\n")
+	m := make(map[string]string)
+	for _, line := range lines {
+		// populate all the network metrics in a map
+		kv := strings.Split(line, ":")
+		if len(kv) != 2 {
+			continue
+		}
+		key := strings.TrimSpace(kv[0])
+		value := strings.TrimSpace(kv[1])
+		m[key] = value
+	}
+
+	receivedBroadcastPackets, err := taskStat.getMapValue(m, "ReceivedBroadcastPackets")
```

Comment:
nit: sorry for not pointing this out earlier, but this code might be less redundant by putting the keys in an array and iterating over it. This is purely a matter of style and not blocker for this PR, so feel free to disregard.

https://play.golang.org/p/1_fArKNzVb6